### PR TITLE
Fixed an issue with SMTP servers which expect a STARTTLS command

### PR DESCRIPTION
**Pull request by siancu**
If the SMTP server listens on port 587 and expects a STARTTLS command before login, then sending the email doesn't work. 

This fixes it.

### CODE REVIEW

File: sendKindle.py [hunk 1]
```diff
@@ -210,7 +217,6 @@ def main():
     '''Run the main program'''
     try:
         kindle = SendKindle()
-        kindle.send_mail()
```
Comment (kparal) [inline sendKindle.py]: Why is this removed?

File: sendKindle.py [hunk 2]
```diff
@@ -169,8 +169,15 @@ def send_mail(self):
...
-            mail_server = smtplib.SMTP_SSL(host=self.smtp_server,
-                                          port=self.smtp_port)
+            if self.smtp_port == 465:
+                mail_server = smtplib.SMTP_SSL(host=self.smtp_server,
+                        port=self.smtp_port)
+            elif self.smtp_port == 587:
+                mail_server = smtplib.SMTP(host=self.smtp_server, port=self.smtp_port)
+                mail_server.starttls()
+            else:
+                raise Exception('port {} not supported'.format(self.smtp_port))
```
Comment (kparal) [inline sendKindle.py]: Why would you restrict the port to just two allowed values? Why not use something like this instead?
```python
if self.smtp_port == 587:
                 mail_server = smtplib.SMTP(...)
                 mail_server.starttls()
else:
                 mail_server = smtplib.SMTP_SSL(host=self.smtp_server,
                        port=self.smtp_port) 
```

File: sendKindle.py (hunk 1 above)
Comment (siancu) [reply sendKindle.py]: By mistake, sorry, I was editing directly the code from pip, and there it seems there's an older version where this function was missing.

File: sendKindle.py (hunk 2 above)
Comment (siancu) [reply sendKindle.py]: Good point, this version is better.

**Comment by siancu** [comment]
> Have you tried whether the new code works with Gmail?

No, sorry, haven't.

**Comment by siancu** [comment]
Hi, I have added a new commit with your suggested changes.

File: sendKindle.py (hunk 1 above)
Comment (kparal) [reply sendKindle.py]: Hmm, there should be the same version:
https://pypi.org/project/sendKindle/#history
No problem.

**Comment by kparal** [comment]
> > Have you tried whether the new code works with Gmail?
> 
> No, sorry, haven't.

What email server have you tried this with? Can you test with gmail?

**Comment by kparal** [comment]
You have to generate an app password:
https://myaccount.google.com/apppasswords

**Comment by siancu** [comment]
I just tried with GMail and it works with both ports 465 and 587 (seems like GMail's SMTP server supports both).

**Comment by kparal** [comment]
OK, thanks
